source/SnapDump-Server/SDSnapshotCall.class.st

```diff
@@ -75,5 +130,25 @@ SDSnapshotCall >> put [
 
 { #category : #public }
 SDSnapshotCall >> snapshot [
-	^ self snapshotWithId: snapshotId 
+	[ (projectName isNil or: [ versionName isNil ])
+		ifTrue: [ 
+			"here we give the possibility to fetch a given snapshot from its snapshotId only.
+			handy but slower than fetching with a known project and version "
+			^ self store snapshotWithId: snapshotId ].
+	^ self store
+		snapshotWithId: snapshotId
+		project: projectName
+		version: versionName ]
+		on: NotFound
+		do: [ :err | ZnRespond signalWith: (ZnResponse notFound: request url) ]
+]
+
+{ #category : #public }
+SDSnapshotCall >> validateRequiredParameters: aCollectionOfParameterNames [
```

Comment:
Why is this here? Does the openApi one not work?

Reply:
I wanted some parameters to be required only for particular call method (#post #head)
I think openApi only allows to make parameters required for all existing call methods, and not for a specific call method, right ?
Anyway I think this will be removed while moving these parameters as path parameters